cuqi/experimental/mcmc/_hmc.py

```diff
@@ -289,32 +305,14 @@ def _pre_warmup(self):
         #  Parameters that does not change during the run
         #    self._mu
 
-        if self._epsilon is None:
-            # parameters dual averaging
-            self._epsilon = self._FindGoodEpsilon()
-            # Parameter mu, does not change during the run
-            self._mu = np.log(10*self._epsilon)
-
-        if self._epsilon_bar is None: # Initial value of epsilon_bar
+        if self._epsilon_bar == "unset": # Initial value of epsilon_bar for tuning
```

Comment:
I could not get rid of pre_warmup and pre_sample in NUTS entirely because there is different behavior for _epsilon_bar depending if you do sample or warmup first. Seems like a strange choice to me. We should perhaps investigate.

Reply:
Agreed!, feel free to create an issue. At some point I'll look at NUTS checkpointing #404 I can also see if there is something we can do about this. The suggestion I mentioned in that stand-up was to update the original NUTS to have more consistent behavior (and add a flag for legacy behavior). Then we can match NUTSNew with the updated NUTS

Reply:
Added it to #404

Reply:
great!